nmdc_schema/dump_single_modality.py

```diff
@@ -0,0 +1,342 @@
+    endpoint_prefix (str): Path component between the URL and the endpoint name.
+    collection_name (str): Name of the collection to fetch documents from.
+    max_docs_per_coll (int): Maximum number of documents to retrieve per collection.
+    page_size (int): Size of each page in pagination.
+
+    Returns:
+    list: List of documents fetched from the collection.
+    """
+    documents = []
+    next_page_token = None
+
+    while len(documents) <= max_docs_per_coll:
+        url = f"{client_base_url}/{collection_name}/{endpoint_prefix}?max_page_size={page_size}"
+
+        if next_page_token:
+            url += f"&page_token={next_page_token}"
+        logger.info(url)
+
+        response = requests.get(url)
+        if response.status_code == 200:
+            data = response.json()
+            resources = data.get("resources", [])
+            documents.extend(resources)
+
+            if "next_page_token" in data:
+                next_page_token = data["next_page_token"]
+            else:
+                break
+        else:
+            logger.error(
+                f"Failed to fetch documents from {collection_name}: {response.status_code} - {response.reason}")
+            break
+
+    return documents
+
+
+# Define the main command group and include the global --schema-source option
+@click.group()
+@click_log.simple_verbosity_option(logger)
+@click.option('--schema-source', required=True, help="The source of the schema. Can be a file path or URL.")
+@click.option('--selected-collections', multiple=True, type=str,
+              help='List of specific collections to fetch data from. If omitted, all collections will be processed.')
+@click.option('--output-yaml', type=click.Path(), required=True,
+              show_default=True, help="Output file for storing fetched data.")
+@click.option('--max-docs-per-coll', default=100, show_default=True,
+              help='Maximum number of documents to retrieve per collection')
+@click.pass_context
+def cli(ctx, schema_source, selected_collections, output_yaml, max_docs_per_coll):
+    """Main command group that handles sub-commands for different access methods."""
+    ctx.ensure_object(dict)
+
+    ctx.obj['SCHEMA_HANDLER'] = SchemaHandler(schema_source)
+    ctx.obj['SELECTED_COLLECTIONS'] = selected_collections
+    ctx.obj['OUTPUT_YAML'] = output_yaml
+    ctx.obj['MAX_DOCS_PER_COLL'] = max_docs_per_coll
+
+
+# Sub-command for API access
+@cli.command()
+@click.option('--client-base-url', default="https://api.microbiomedata.org", show_default=True,
+              help='HTTP(S) path to the API server.')
+@click.option('--endpoint-prefix', default="nmdcschema", show_default=True,
+              help='API path component between the URL and the endpoint name')
+@click.option('--page-size', default=200_000, show_default=True,
+              help='Size of each page in pagination.')
+@click.pass_context
+def api_access(ctx, client_base_url, endpoint_prefix, page_size):
```

Comment:
Is there a clearer verb-noun type of name for these functions?  `dump_from_api` / `dump_from_db` or something like that

Reply:
Thanks. See below.